ftl/common/builder_runner.py

```diff
@@ -91,37 +91,71 @@ def GetCachedDepsImage(self, checksum):
             logging.info('Storing layer cash.')
-            self.cash.Store(self.args.base, self.builder.namespace, checksum,
+            self.cash.Store(self.base, self.builder.namespace, checksum,
                             dep_image)
         else:
             logging.info('Skipping storing layer cash.')
 
     def GenerateFTLImage(self):
         with docker_image.FromRegistry(self.base_name, self.base_creds,
-                                       self.transport) as self.args.base:
-
+                                       self.transport) as self.base:
             # Create (or pull from cache) the base image with the
             # package descriptor installation overlaid.
-            logging.info('Generating dependency layer...')
-            checksum = self.GetCacheKey(self.builder.descriptor_files)
-            deps_image = self.GetCachedDepsImage(checksum)
-            if not deps_image:
-                # TODO(aaron-prindle) make this better, prob pass args to bldr
-                extracted = _args_extractor(vars(self.args))
-                deps_image = self.builder.CreatePackageBase(**extracted)
-                self.StoreDepsImage(deps_image, checksum)
-            # Construct the application layer from the context.
-            logging.info('Generating app layer...')
-            app_layer, diff_id = self.builder.BuildAppLayer()
-            with append.Layer(
-                    deps_image, app_layer, diff_id=diff_id) as app_image:
-                if self.args.output_path:
+            # ftl_image = self.base
+            with ftl_util.Timing("generating_interpreter_layer"):
+                interpreter_key = self.builder.interpreter_parser()
+                checksum = self.GetCacheKey(interpreter_key)
+                # GetCachedDepsImage returns self.base if checksum is None
+                interpreter_image = self.GetCachedDepsImage(checksum)
+                if not interpreter_image:
+                    interpreter_layer, diff_id, overrides = \
+                        self.builder.BuildInterpreterLayer()
+                    interpreter_image = append.Layer(
+                        self.base,
+                        interpreter_layer,
+                        diff_id=diff_id,
+                        overrides=overrides)
+                    self.StoreDepsImage(interpreter_image, checksum)
+
+            with ftl_util.Timing("generating_dependency_layer"):
+                pkg_txts = self.builder.pkg_txt_parser(
+                    self.builder.descriptor_files, self.ctx)
+                pkg_keys = self.builder.pkg_key_parser(
+                    self.builder.descriptor_files, self.ctx)
+
+                if len(pkg_txts) != len(pkg_keys):
+                    # TODO(aaron-prindle) verify best way to exit
+                    exit(1)
+                checksums = self.GetCacheKeys(pkg_keys)
+                # TODO(aaron-prindle) parallelize this
+                logging.info("===================")
```

Comment:
Did you mean to leave these logging calls in here?

Reply:
no, removed